src/NuGet.Clients/PackageManagement.VisualStudio/ProjectSystems/LegacyCSProjPackageReferenceProjectProvider.cs

```diff
@@ -52,6 +65,9 @@ public bool TryCreateNuGetProject(EnvDTE.Project dteProject, ProjectSystemProvid
                 return false;
             }
 
+            // Lazy load the CPS enabled JoinableTaskFactory for the UI.
+            NuGetUIThreadHelper.SetJoinableTaskFactoryFromService(_componentModel.Value);
```

Comment:
It can still calls into CPS API's as there could be some projects which are CPS based but not .net core projects so they will fall into Legacy csproj so to save guard against those projects, i added this.

But `GetUnconfiguredProject ` won't be called in this flow, so if you think there won't be any other way for project systems to call into CPS then i can remove it.